core/src/main/scala/transport/netty/ClientConnectionPool.scala

```diff
@@ -77,6 +78,7 @@ class NettyConnectionPool(hosts: Process[Task,InetSocketAddress],
         }
       }
       else {
+        pipe.removeLast()
         channel.close()
```

Comment:
I that would be included then I suppose it should be done like that but also have:
```scala
 ... } flatmap {case _ => Task.fail(IncompatibleServer("server missing required capabilities: " + missing)) }
```
However, I am a bit ignorant when it comes to the netty stuff wrt. safety and whether a simple `channel.close` is ok in this scenario.